pkg/stub/handler.go

```diff
@@ -372,10 +472,29 @@ func (h *Handler) Handle(ctx context.Context, event sdk.Event) error {
 		if srcfg.Name != v1alpha1.SamplesResourceName {
 			return nil
 		}
+
+		// pattern is 1) come in with delete timestample, event delete flag false
+		// 2) then after we remove finalizer, comes in with delete timestamp
+		// and event delete flag true
 		if event.Deleted {
-			logrus.Println("SampleResource deleted")
-			h.samplesResource = nil
-			return h.CreateDefaultResourceIfNeeded()
+			// possibly tell poller to stop
+			logrus.Info("A previous delete attempt has been successfully completed")
+			return nil
+		}
+		if srcfg.DeletionTimestamp != nil {
+			if h.NeedsFinalizing(srcfg) {
+				logrus.Println("Initiating finalizer processing for a SampleResource delete attempt")
+				h.RemoveFinalizer(srcfg)
+				err := h.GoodConditionUpdate(srcfg, corev1.ConditionFalse, v1alpha1.SamplesExist)
```

Comment:
i think this should be done before removing the finalizer.  You want to complete all the work you want to do related to the deletion of the object, before you remove the finalizer.

Reply:
the `h.GoodConditionUpdate` is what persists the changes from `h.RemoveFinalizer` to etcd ... `h.RemoveFinalizer` just changes the local copy

Reply:
ok